sheer/indexer.py

```diff
@@ -60,6 +60,77 @@ def mapping(self, default_mapping):
             return copy.deepcopy(default_mapping)
 
 
+def index_document(es, index_name, processor, document):
+    """
+    Create the given document from the given content processor in the
+    given Elasticsearch instance for the  given index.
+
+    If the document already exists in Elasticsearch it will be updated.
+    """
+    # Create the document. If it already exists in Elasticsearch an
+    # exception will be raised.
+    try:
+        es.create(index=index_name,
+                doc_type=processor.name,
+                id=document['_id'],
+                body=document)
+    except TransportError, e:
+        # Elasticsearch status code 409 is DocumentAlreadyExistsException
+        # Anything else and we want to bail here.
+        if e.status_code != 409:
+            raise e
+
+        # If the document couldn't be created because it already exists,
+        # update it instead.
+        es.update(index=index_name,
+                doc_type=processor.name,
+                id=document['_id'],
+                body={'doc': document})
+
+
+def index_processor(es, index_name, default_mapping, processor, reindex=False):
+    """
+    Index all the documents provided by the given content processor for
+    the given index in the given Elasticsearch instance.
+
+    If reindex=True and the processor already exists the mapping in
+    Elasticsearch will be destroyed and recreated and all documents will
+    be created anew.
+    """
+    # If the mapping already exists, and we were called with the reindex
+    # flag, remove the mapping.
+    mapping = es.indices.get_mapping(index=index_name, doc_type=processor.name)
+    if mapping and reindex:
+        print "removing existing mapping for %s (%s)" % (processor.name, processor.processor_name)
+        es.indices.delete_mapping(index=index_name, doc_type=processor.name)
+        mapping = {}
+
+    # Then create the mapping if it does not exist
+    if not mapping:
+        print "creating mapping for %s (%s)" % (processor.name, processor.processor_name)
+        es.indices.put_mapping(index=index_name,
+                            doc_type=processor.name,
+                            body={processor.name: processor.mapping(default_mapping)})
+
+    # Get the document iterator from the processor.
+    # This could raise a value error raised by json.loads() if there is a
+    # connection error, server error, or other error acquiring the data.
+    try:
+        document_iterator = enumerate(processor.documents())
+    except ValueError:
+        # print an error message and raise the exception. It should be caught by
+        # this function's caller.
+        sys.stderr.write("error reading documents for %s" % processor.name)
+    else:
```

Comment:
This is where the error is written to stderr. I changed this to be a `try`/`except`/`else` so that we don't have the `try`/`except` again in the for loop in `index_location()`.